### PR TITLE
Add the Colorado HB24-1311 Family Affordability Tax Credit

### LINKED ISSUE #4367: Add the Colorado HB24-1311 Family Affordability Tax Credit
[Reference](https://leg.colorado.gov/bills/hb24-1311)

[Bill](https://leg.colorado.gov/sites/default/files/documents/2024A/bills/2024a_1311_01.pdf)

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #4367

Co-authored-by: RileyAK <RileyAK@users.noreply.github.com>

### CODE REVIEW

File: policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        def formula(person, period, parameters):
+            age = person("age", period)
+            depenent = person("is_child_dependent", period)
+            p = parameters(
+                period
+            ).gov.contrib.co_general_assembly.family_affordability_credit
+            base_amount = p.amount.calc(age) * depenent
+            agi = person.tax_unit("adjusted_gross_income", period)
+            filing_status = person.tax_unit("filing_status", period)
+            phase_out_start = where(
```
Comment (MaxGhenis) [inline policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py:23]: please add comments to explain this non-obvious code

File: policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        entity = Person
+        label = "Colorado Family Affordability Credit"
+        unit = USD
+        definition_period = YEAR
+        reference = "https://leg.colorado.gov/bills/hb24-1311"
+        defined_for = StateCode.CO
+
+        def formula(person, period, parameters):
+            age = person("age", period)
+            depenent = person("is_child_dependent", period)
```
Comment (MaxGhenis) [inline policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py:16]:
```suggestion
            dependent = person("is_child_dependent", period)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]
please also add a README.md to the co_general_assembly parameter folder

File: policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    class co_refundable_credits(Variable):
+        value_type = float
+        entity = TaxUnit
+        label = "Colorado refundable credits"
+        unit = USD
+        definition_period = YEAR
+        defined_for = StateCode.CO
+
+        def formula(tax_unit, period, parameters):
+            old_credits = add(
```
[the remark this answers is not in the document]
Comment (PavelMakarchuk) [reply policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py:52]: I think we need we need an add function if we want to access list parameter values

File: policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py (hunk 1 above)
Comment (PavelMakarchuk) [reply policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py:25]: added

File: policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (MaxGhenis) [reply policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py:54]: this is still an add function

File: policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py (hunk 1 above)
Comment (MaxGhenis) [reply policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py:25]: Oh I see, the first boundary is the young child age limit. Please create a new python variable for it to clarify and reduce duplication.

Comment (PavelMakarchuk) [reply policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py:26]: done

File: policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            dependent = person("is_child_dependent", period)
+            p = parameters(
+                period
+            ).gov.contrib.co_general_assembly.family_affordability_credit
+            base_amount = p.amount.calc(age) * dependent
+            agi = person.tax_unit("adjusted_gross_income", period)
+            filing_status = person.tax_unit("filing_status", period)
+            # The phase-out amounts, start points, and increments are differently defined
+            # for younger and older dependents.
+            young_child_age_limit = age < p.amount.thresholds[1]
```
Comment (MaxGhenis) [inline policyengine_us/reforms/co_general_assembly/co_family_affordability_credit.py:25]:
```suggestion
            young_child = age < p.amount.thresholds[1]
```

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**